### DISCUSSION TITLE
Table of Contents vs Summary section

### DISCUSSION THREAD

**Discussion by seancorfield** [bbatsov/clojure-style-guide#218] [Ideas]
@bbatsov asked in #216 whether the TOC satisfied my request for a distinct "bullet point" summary of the guidelines.

The TOC items indicate what _type_ of guidelines there are but not the _actual_ guidelines themselves. For example "Link Breaks in ns" just tells you that the guideline is _about_ newlines in `ns` but gives no indication of whether you should or should not have them nor of when you should have them. The TOC gives you an idea of what guidelines there are and lets you get to the guideline(s) for a specific topic, but you still have to go to each section, in turn, to see what the guidelines actually are.

I agree that duplicating content introduces a maintenance overhead but there is currently no way to quickly read over just the guidelines themselves -- you have to wade through the whole document, picking them out of the rationale and examples. I think having an easy-to-read summary at the start of the document is worth that extra maintenance.

**Comment by bbatsov** [comment]
I should clarify that the titles are pretty bad currently, as post of them were auto-generated from the anchors that guidelines used to have before the conversion to the current format. Here's the commit for reference https://github.com/bbatsov/clojure-style-guide/commit/798001a76262d854cb68eb2cc59caa21094d20f9 It'd be nice if we improved the titles to be both more descriptive and more consistent. I guess we'll have to do this at some point. 

> I agree that duplicating content introduces a maintenance overhead but there is currently no way to quickly read over just the guidelines themselves -- you have to wade through the whole document, picking them out of the rationale and examples. I think having an easy-to-read summary at the start of the document is worth that extra maintenance.

Fair enough. There might be some way to generate a TOC for each section to avoid having to do this manually.
